tracing/tests/enabled.rs

```diff
@@ -10,21 +10,30 @@ use tracing::Level;
 #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
 #[test]
 fn level_and_target() {
-    let (collector, handle) = collector::mock()
+    let collector = collector::mock()
         .with_filter(|meta| {
             if meta.target() == "debug_module" {
                 meta.level() <= &Level::DEBUG
+            } else if meta.is_span() {
+                meta.level() <= &Level::TRACE
+            } else if meta.is_event() {
+                meta.level() <= &Level::DEBUG
             } else {
                 meta.level() <= &Level::INFO
             }
         })
         .done()
-        .run_with_handle();
+        .run();
 
     tracing::collect::set_global_default(collector).unwrap();
 
     assert!(tracing::enabled!(target: "debug_module", Level::DEBUG));
     assert!(tracing::enabled!(Level::ERROR));
     assert!(!tracing::enabled!(Level::DEBUG));
-    handle.assert_finished();
+
+    // Ensure that the `_event` and `_span` alternatives work corretly
+    assert!(!tracing::event_enabled!(Level::TRACE));
+    assert!(tracing::event_enabled!(Level::DEBUG));
+
+    assert!(tracing::span_enabled!(Level::TRACE));
```

Comment:
personally, i would probably have split these out into their own separate tests. not a blocker.

we should probably also test these macros with overridden targets.